uploader.go

```diff
@@ -269,10 +303,6 @@ func Upload(options UploadOptions) (<-chan error, error) {
 		return nil, errors.New("no client provided")
 	}
 
-	if options.GetKey == nil {
```

Comment:
@skeggse thoughts on keeping `GetKey` as an upload option, but changing it's signature to
`GetKey(time.Time, string /* objectType */)`? The benefit there is a little less redundancy (passing around both key and object type), and perhaps more uniformity in key structure (though that's up to the client)

Reply:
Hmmm yeah that'd be fine by me.